lib/modules/manager/bazel-module/fragments.ts

```diff
@@ -0,0 +1,269 @@
+import is from '@sindresorhus/is';
+import { z } from 'zod';
+import * as starlark from './starlark';
+
+// Fragment Schemas
+
+const FragmentTypeSchema = z.enum([
```

Comment:
Use [discriminated unions](https://www.npmjs.com/package/zod#discriminated-unions)

to simplify the code base.

Reply:
I don't know how to leverage discriminated unions in this case. Did you have something in mind?

For context, I introduced `CommonFragmentSchema`, which includes the `type: FragementTypeSchema`. The `CommonFragmentSchema` is the reference type for fragments that can refer to other fragments. This allows `ArrayFragmentSchema` and `AttributeFragmentSchema` to refer to any other fragment.

Reply:
Just noticed the [recursive type support](https://github.com/colinhacks/zod#recursive-types). I will give that a try.

Reply:
@secustor While trying out some solutions, I came across [this issue, starting a discussion about deprecating discriminatedUnion](https://github.com/colinhacks/zod/issues/2106). I wonder if we should reconsider converting this PR to use discriminatedUnion. Thoughts?

Reply:
It looks like the jury is still out regarding discrimnatedUnion vs the new switch capability. 🤷‍♂️

Reply:
I implemented a discriminatedUnion.